src/scroll-timeline-base.js

```diff
@@ -132,155 +136,41 @@ function resolvePx(cssValue, resolvedLength) {
-    // to resolve a scroll timeline offset for scroll offset with the is
-    // first flag set to first offset.
-    let effectiveOffset = calculateScrollOffset(
-      firstOffset ? new CSSUnitValue(0, 'percent') : new CSSUnitValue(100, 'percent'),
-      source,
-      orientation,
-      scrollOffsets[i],
-      fns[i]);
-    //  5.2 If effective offset is null, the effective scroll offsets is empty and abort the remaining steps.
-    if(effectiveOffset === null)
-      return [];
-    // 5.3 Add effective offset into effective scroll offsets.
-    effectiveScrollOffsets.push(effectiveOffset);
-    // 5.4 Set first offset to false.
-    firstOffset = false;
+
+  const display  = getComputedStyle(node).display;
+  if (display == 'none') {
+    updateSource(timeline, null);
+    return;
   }
-  // 6. Return effective scroll offsets.
-  return effectiveScrollOffsets;
+
+  const source = getScrollParent(node.parentNode);
+  updateSource(timeline, source);
 }
 
-/**
- * Compute scroll timeline progress per
- * https://drafts.csswg.org/scroll-animations-1/#progress-calculation-algorithm
- * @param offset {number}
- * @param scrollOffsets {Array}
- * @returns {number}
- */
-export function ComputeProgress(
-  offset,
-  scrollOffsets
-) {
-  // 1. Let scroll offsets be the result of applying the procedure to resolve
-  // scroll timeline offsets for scrollOffsets.
-  // 2. Let offset index correspond to the position of the last offset in
-  // scroll offsets whose value is less than or equal to offset and the value
-  // at the following position greater than offset.
-  let offsetIndex;
-  for (offsetIndex = scrollOffsets.length - 2;
-       offsetIndex >= 0 && 
-         !(scrollOffsets[offsetIndex] <= offset && offset < scrollOffsets[offsetIndex + 1]);
-       offsetIndex--) {
-  }
-  // 3. Let start offset be the offset value at position offset index in
-  // scroll offsets.
-  let startOffset = scrollOffsets[offsetIndex];
-  // 4. Let end offset be the value of next offset in scroll offsets after
-  // start offset.
-  let endOffset = scrollOffsets[offsetIndex + 1];
-  // 5. Let size be the number of offsets in scroll offsets.
-  let size = scrollOffsets.length;
-  // 6. Let offset weight be the result of evaluating 1 / (size - 1).
-  let offsetWeight = 1 / (size - 1);
-  // 7. Let interval progress be the result of evaluating
-  // (offset - start offset) / (end offset - start offset).
-  let intervalProgress =  (offset - startOffset) / (endOffset - startOffset);
-  // 8. Return the result of evaluating
-  // (offset index + interval progress) × offset weight.
-  return (offsetIndex + intervalProgress) * offsetWeight;
+function updateSource(timeline, source) {
+  const oldSource = scrollTimelineOptions.get(timeline).source;
+  if (oldSource == source)
+    return;
+
+  const listener = () => {
+    updateInternal(timeline);
+  };
+  if (oldSource)
+    scrollEventSource(oldSource).removeEventListener("scroll", listener);
```

Comment:
I don't think this will work, isn't listener different in every call to updateSource? I think you'd need to keep the old bound listener.

Reply:
Agreed. Would need to reference count to determine if we need to keep the old listener around since we could have multiple scroll-linked animations.  This logic was moved from ScrollTimeline's setter for source.  I can update to only remove if the animation list becomes empty.

Reply:
done